image/roxctl/rhtap.Dockerfile

```diff
@@ -12,6 +12,17 @@ WORKDIR /go/src/github.com/stackrox/rox/app
 
 COPY . .
 
+# When dependencies prefetch is on, RHTAP modifies this Dockerfile which would fail the following check.
+# This should stop happening after https://issues.redhat.com/browse/STONEBLD-1847
+# Since changes to the Dockerfile aren't needed (in its copy inside the build context), we revert changes.
+# This is only done when in RHTAP which is checked by the presence of cachi2.env file in absence of anything better.
+RUN if [[ -f /cachi2/cachi2.env ]]; then git restore image/roxctl/rhtap.Dockerfile; fi
+
+RUN echo "Checking that files in git are not modified." && \
```

Comment:
Have you tried combining this in one layer (ie one RUN)? 
Since we likely have to repeat this workaround for at least `main` image, it might make sense to preemptively put it in a script.

Reply:
> Have you tried combining this in one layer (ie one RUN)?

Reducing number of `RUN`s does not win us much in the `builder` stage because it's not the one users pull. Unless you have in mind benefit other than reducing the number of pulled layers?

I'm creating a script but have difficulties finding the optimal place for it in the repo. Do you have any suggestions?

Reply:
> Unless you have in mind some benefit other than reducing the number of pulled layers?
You're right, the number of builder stages should not matter. 

> finding the optimal place for it in the repo
`scripts/ci/rhtap`, `.rhtap`, `.tekton/scripts` might be options

Reply:
Used `scripts/rhtap/fail-build-if-git-is-dirty.sh`.
Did not use `scripts/ci` because files there looked very much intended for CI. This file here is executed in Dockerfile which will be built not necessarily in CI (at least we intend to).